ilastik/applets/counting/countingGuiBoxesInterface.py

```diff
@@ -626,51 +623,36 @@ def release(self):
 
 
 class BoxInterpreter(QObject):
-    leftClickReleased = pyqtSignal(QRect)
+    boxDrawn = pyqtSignal(QRect)
 
-    def __init__(self, navigationInterpreter, positionModel, BoxContr, widget):
-        """
-        Class which interacts directly with the image scene
-
-        :param navigationInterpreter:
-        :param positionModel:
-        :param BoxContr:
-        :param widget: The main widget
-
-        """
-
-        QObject.__init__(self)
+    def __init__(self, base: NavigationInterpreter, posModel: PositionModel, mainWidget: QWidget):
```

Comment:
not sure how `mainWidget` is more descriptive than just widget, but after consulting `QRubberBand` docs I'm not wiser ;). The Qt sourcecode gives me the feeling that it should be called `parent`

Reply:
I'm afraid that calling this parameter `parent` would give the wrong impression that `BoxInterpreter` itself has a notion of a "parent" widget, something that `QObject` does.
How about `centralWidget`, just to be consistent with `LayerViewerGui`?

Reply:
okay, but maybe add a docstring making clear that it's just the parent for the rubberwidget.

Reply:
I ended up renaming it to `newBoxParent` with the corresponding docstring.